### PR TITLE
Fix NRE in AnalyzerConfigOptions computation

### LINKED ISSUE #3613: ReviewSqlQueriesForSecurityVulnerabilities threw an exception of type 'System.ArgumentNullException'
#### Analyzer package

Unknown

#### Package Version

3.3.0-beta1.20257.5

#### Diagnostic ID

CA2100

#### Repro steps

Use in dotnet/roslyn@27654b0.

#### Actual behavior

        vbc error AD0001: Analyzer 'Microsoft.NetCore.Analyzers.Data.ReviewSqlQueriesForSecurityVulnerabilities' threw an exception of type 'System.ArgumentNullException' with message 'Value cannot be null. [C:\dev\roslyn\src\Compilers\VisualBasic\Portable\Microsoft.CodeAnalysis.VisualBasic.vbproj]
    Parameter name: key'.
    Exception occurred with following context:
    Compilation: Microsoft.CodeAnalysis.VisualBasic
    ISymbol:  (Namespace)
    System.ArgumentNullException: Value cannot be null.
    Parameter name: key
       at System.Collections.Immutable.Requires.FailArgumentNullException(String parameterName)
       at System.Collections.Immutable.ImmutableDictionary`2.TryGetValue(TKey key, TValue& value)
       at Analyzer.Utilities.AggregateCategorizedAnalyzerConfigOptions.TryGetOptionValue[T](String optionName, SyntaxTree tree, DiagnosticDescriptor rule, TryParseValue`1 tryParseValue, T defaultValue, T& value)
       at Analyzer.Utilities.AggregateCategorizedAnalyzerConfigOptions.GetOptionValue[T](String optionName, SyntaxTree tree, DiagnosticDescriptor rule, TryParseValue`1 tryParseValue, T defaultValue)
       at Analyzer.Utilities.AnalyzerOptionsExtensions.GetSymbolNamesWithValueOption[TValue](AnalyzerOptions options, String optionName, DiagnosticDescriptor rule, SyntaxTree tree, Compilation compilation, CancellationToken cancellationToken, String namePrefixOpt, String optionDefaultValue, String optionForcedValue, Func`2 getTypeAndSuffixFunc)
       at Analyzer.Utilities.AnalyzerOptionsExtensions.GetExcludedSymbolNamesWithValueOption(AnalyzerOptions options, DiagnosticDescriptor rule, ISymbol symbol, Compilation compilation, CancellationToken cancellationToken)
       at Analyzer.Utilities.Extensions.ISymbolExtensions.IsConfiguredToSkipAnalysis(ISymbol symbol, ISymbol containingContextSymbol, AnalyzerOptions options, DiagnosticDescriptor rule, Compilation compilation, CancellationToken cancellationToken)
       at Microsoft.NetCore.Analyzers.Data.ReviewSqlQueriesForSecurityVulnerabilities.<>c__DisplayClass7_0.<Initialize>b__1(OperationBlockStartAnalysisContext operationBlockStartContext)
       at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__62`6.<ExecuteBlockActionsCore>b__62_2(ValueTuple`5 data)
       at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
    -----
    Suppress the following diagnostics to disable this analyzer: CA2100

### PR DESCRIPTION
**Pull request by mavasani**
Gracefully handle requests to compute options for unsupported symbol kinds
Fixes #3613

### CODE REVIEW

File: src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs [hunk 1]
```diff
@@ -21,14 +22,38 @@ internal static partial class AnalyzerOptionsExtensions
         private static readonly ImmutableHashSet<OutputKind> s_defaultOutputKinds =
             ImmutableHashSet.CreateRange(Enum.GetValues(typeof(OutputKind)).Cast<OutputKind>());
 
+        private static bool TryGetSyntaxTreeForOption(ISymbol symbol, [NotNullWhen(returnValue: true)] out SyntaxTree? tree)
+        {
+            switch (symbol)
```
Comment (sharwell) [inline src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs:27]: 💡 Switching over the symbol _kind_ will be more efficient than interface type checks.

Comment (mavasani) [reply src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs]: Done, thanks!

File: src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs [hunk 2]
```diff
@@ -21,14 +22,38 @@ internal static partial class AnalyzerOptionsExtensions
...
+        private static bool TryGetSyntaxTreeForOption(ISymbol symbol, [NotNullWhen(returnValue: true)] out SyntaxTree? tree)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Assembly:
+                    tree = null;
+                    return false;
+
+                case SymbolKind.Namespace:
+                    if (((INamespaceSymbol)symbol).IsGlobalNamespace)
```
Comment (sharwell) [inline src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs:34]:
```suggestion
                case SymbolKind.Namespace when (((INamespaceSymbol)symbol).IsGlobalNamespace):
```

**Comment by mavasani** [comment]
Ping @sharwell

**Comment by mavasani** [comment]
Ping @sharwell

**Comment by mavasani** [comment]
Thanks @sharwell!

**Pull request merged**